pydantic/main.py

```diff
@@ -620,6 +617,7 @@ def validate_model(  # noqa: C901 (ignore complexity)
 
     if check_extra:
         extra = input_data.keys() - names_used
+        fields_set |= extra
```

Comment:
you can put this one line down inside `if extra`. Or even inside `if config.extra is Extra.allow:`